operator/objectsusercontroller/connector.go

```diff
@@ -56,33 +54,43 @@ func (c *objectsUserConnector) Connect(ctx context.Context, mg resource.Managed)
 
 // createCloudscaleClient creates a new kube using the API token provided.
 func (c *objectsUserConnector) createCloudscaleClient(ctx context.Context) *cloudscalesdk.Client {
-	tc := oauth2.NewClient(ctx, oauth2.StaticTokenSource(&oauth2.Token{AccessToken: c.apiToken}))
+	token := pipeline.MustLoadFromContext(ctx, apiTokenKey{}).(string)
```

Comment:
This type assertion could panic, if the returned value is `nil` or not a string.

This will avoid panics, but the ok value would need to be checked:
```suggestion
	token, ok := pipeline.MustLoadFromContext(ctx, apiTokenKey{}).(string)
```

Reply:
Yes, it could panic. But they are set in previous pipeline steps, and if the previous steps fail, then it won't get executed.
We use this kind of object retrieval in various places already.
In these cases I favor readable business code above type assertion handling with more `if !ok || var == nil`s. If they panic given a non-existing or wrong type than I rather let it crash than silently just logging it. Clearly a case as this would be a programmer fault for not testing the order of steps.

I could certainly check this but the code will get more ugly in many places...

Reply:
So the pipeline ensures that there's no faulty value here at this point? If so then it's fine by me.